### PR TITLE
[NO-TICKET] Refactoring

### PR DESCRIPTION
**Pull request by vekaev**
- [ ] Add full test coverage
- [x] Fix mistakes in README.md
- [x] Add eslint, prettier, husky, lint-staged
- [x] Add pre-push script before commit to bundle ts files and docs
- [x] Add pre-commit script before commit to lint ts files
- [x] Add npmrc and engines to package.json
- [x] Rewrite prototype to class, change _ to private methods
- [x] Change to ESM modules import/export
- [x] Add build command to workflows
- [x] Remove devDependencies from dependencies
- [x] Update packages
- [x] Remove documentations from deployed npm package
- [ ] Add missing types for all methods
- [ ] Add Mock Service Worker to test requests
- [ ] ??Create ClientV2 class with w2 methods

### CODE REVIEW

File: .DS_Store
Comment (alejouribesanchez) [inline .DS_Store:1]: Add this file to the .gitignore

File: docs/.DS_Store
Comment (alejouribesanchez) [inline docs/.DS_Store:1]: Add this file to gitignore

**Review by alejouribesanchez — CHANGES_REQUESTED** [review]
Add .DS_STORE files to gitignore

**Comment by vekaev** [comment]
lol)

**Comment by alejouribesanchez** [comment]
Conflicts

**Pull request closed without merging**